php/php.editor/src/org/netbeans/modules/php/editor/codegen/CGSInfo.java

```diff
@@ -369,17 +395,40 @@ private String getFirstTypeFromBlock(final PHPDocBlock phpDoc) {
         }
 
         private String getFirstTypeFromTag(final PHPDocTypeTag typeTag) {
+            boolean canBeNull = canBeNull(typeTag);
             String result = ""; //NOI18N
             for (PHPDocTypeNode typeNode : typeTag.getTypes()) {
                 String type = typeNode.getValue();
-                if (!Type.isPrimitive(type) && !VariousUtils.isSpecialClassName(type)) {
+                if (phpVersion.compareTo(PhpVersion.PHP_70) >= 0
+                        && !VariousUtils.isSpecialClassName(type)
+                        && !Type.isInvalidPropertyType(type)) {
+                    result = typeNode.isArray() ? Type.ARRAY : type;
+                    if (canBeNull && phpVersion.compareTo(PhpVersion.PHP_71) >= 0) {
```

Comment:
Consider adding e.g. hasNullableTypes() directly to PhpVersion. It would be much easier to use and to read.